crates/prover/src/tls/config.rs

```diff
@@ -22,11 +22,21 @@ pub struct ProverConfig {
     #[builder(default = "DEFAULT_MAX_SENT_LIMIT")]
     max_sent_data: usize,
     /// Maximum number of bytes that can be decrypted online.
-    #[builder(default = "DEFAULT_MAX_RECV_LIMIT")]
+    #[builder(default = "0")]
     max_recv_data_online: usize,
     /// Maximum number of bytes that will be decrypted after the TLS connection is closed.
-    #[builder(default = "0")]
+    #[builder(default = "DEFAULT_MAX_RECV_LIMIT")]
     max_deferred_size: usize,
+    /// Defers the decryption from the start of the connection.
+    ///
+    /// Decryption of responses will be deferred until after the TLS connection is closed. This is
+    /// useful if you either have only one request/response cycle of if you have several such
+    /// cycles but the content of the request never depends on the content of the previous response.
+    ///
+    /// This allows to decrypt responses locally without MPC, so this option saves bandwidth
+    /// and performance.
+    #[builder(default = "true")]
+    defer_decryption_from_start: bool,
```

Comment:
I feel it would be better to refer to when it is deferred to (after close), instead of saying it is deferred at the start

Reply:
Hmm I explicitly added `from_start` because we can also enable it midflight, so that people do not wonder what this is for when we also have `defer_decryption()`.